game_patch/multi/multi.cpp

```diff
@@ -293,20 +320,25 @@ bool multi_is_weapon_fire_allowed_server_side(rf::Entity *ep, int weapon_type, b
     rf::Player* pp = rf::player_from_entity_handle(ep->handle);
     if (ep->ai.current_primary_weapon != weapon_type) {
         xlog::debug("Player {} attempted to fire unselected weapon {}", pp->name, weapon_type);
+        send_private_message_for_cancelled_shot(pp, "You attempted to fire an unselected weapon.");
```

Comment:
Firing while dying triggers this code in your server currently. If correct maybe check whether entity's state is not dying.